Sources/XCHammer/InstallVFS.swift

```diff
@@ -0,0 +1,141 @@
+                                    roots: vfsFile.roots) else {
+        print("warning: no framwork entry - skiping copy files for framework \(frameworkName)")
+        return
+    }
+
+    // Note: This could be asserted works for rules_ios - see other comment in
+    // XcodeTarget.swift relating to it. This breaks for .apps right now and
+    // hits other problems
+    guard let fwRoot = findRoot(name: "\(frameworkName).framework", roots:
+                                frameworks.contents ?? []) else {
+        print("warning: Can't find framework \(frameworkName)")
+        return
+    }
+
+    // Resolve subset of roots relative to the VFS
+    FileManager.default.changeCurrentDirectoryPath(dirname(vfsPath))
+
+    let dstFrameworkPath = targetBuildDir + Path(frameworkName + ".framework")
+
+    // Consider unifying these 3 looping bodies after addressing the comments
+    if let headers = findRoot(name: "Headers", roots: fwRoot.contents ?? []) {
+        try? FileManager.default.createDirectory(atPath: (dstFrameworkPath + Path("Headers")).string,
+                    withIntermediateDirectories: true,
+                    attributes: [:])
+        headers.contents?.forEach  { header in
+            guard let headerPathS = header.externalContents  else {
+                return
+            }
+            let headerPath = Path(headerPathS)
+            let fwHeader = dstFrameworkPath + Path("Headers") + Path(headerPath.lastComponent)
+            try? FileManager.default.createSymbolicLink(at: fwHeader.url, 
+                                    withDestinationURL: headerPath.url)
+        }
+    }
+
+    if let privateHeaders = findRoot(name: "PrivateHeaders", roots: fwRoot.contents ?? []) {
+        try? FileManager.default.createDirectory(atPath: (dstFrameworkPath + Path("PrivateHeaders")).string,
+                    withIntermediateDirectories: true,
+                    attributes: [:])
+        privateHeaders.contents?.forEach  { header in
+            guard let headerPathS = header.externalContents  else {
+                return
+            }
+            let headerPath = Path(headerPathS)
+            let fwHeader = dstFrameworkPath + Path("PrivateHeaders") + Path(headerPath.lastComponent)
+            try? FileManager.default.createSymbolicLink(at: fwHeader.url, 
+                                    withDestinationURL: headerPath.url)
+        }
+    }
+
+    try? FileManager.default.createDirectory(atPath: (dstFrameworkPath + Path("Modules")).string,
+                withIntermediateDirectories: true,
+                attributes: [:])
+
+    if let modules = findRoot(name: "Modules", roots:
+                                fwRoot.contents ?? []) {
+        modules.contents?.forEach  { header in
+            guard let headerPathS = header.externalContents  else {
+                return
+            }
+            // FIXME: This is wrong! use the name before merging this
```

Comment:
- [ ] Still TBD before merging?

Reply:
I don't know if I'm going to have bandwidth to get to this today, I think it's fine for now. We should probably re-visit this run-script longer term